BotModules/CommanderInChief/commander-in-chief.js

```diff
@@ -0,0 +1,174 @@
+}
+
+// Registers the basic help command
+function registerHelp() {
+  registerCommand(undefined, {
+    keyword: '/help',
+    helptext: 'When you are a noob and need to reference the commands',
+    action: help,
+    args: [{
+      index: 1,
+      helptext: 'Include context to filter help commands displayed. "/help <context>" will display the command documentation for that context. Call "/help all" to see all commands or "/help list" to list all contexts.'
+    }]
+  });
+}
+
+//Help text generation. Appears in the following format:
+//
+// uses code block tricks to format the text - https://gist.github.com/matthewzring/9f7bbfd102003963f9be7dbcf7d40e51
+//  - msg: the message object from discord https://discord.js.org/#/docs/main/stable/class/Message
+function help(msg, args){
+  var reply = `Looks like ${msg.member} is an idiot and needs guidence on the commands: \n\n`;
+  var filterContext = args.raw[1] ? args.raw[1].toUpperCase() : undefined;
+
+  var contexts = Array.from(new Set(commands.map(command => command.context))).sort();
+
+  switch(filterContext){
+    case 'ALL':
+      printAll();
+      break;
+    case 'LIST':
+      printListOfContexts();
+      break;
+    default:
+      printContext(filterContext);
+      break;
+  }
+
+  function printAll(){
+    printContext(undefined);
+
+    contexts.forEach(context => {
+      if(context) printContext(context);
+    });
+  }
+
+  function printListOfContexts(){
+    reply += "```asciidoc" + `\n`;
+    reply += "Contexts" + `\n`;
+    reply += "=====" + `\n`;
+    contexts.forEach(context => {
+      if(context) reply += "- " + context + `\n`;
+    });
+    reply += "```" + `\n`;
+  }
+
+  function printContext(context){
+    reply += "```asciidoc" + `\n`;
+    if(context) reply += context + `\n`;
+    if(context) reply += "=====" + `\n`;
+    //else        reply += "Base Commands: " + `\n`;
+    commands.filter(command => CaseInsensitive(command.context) == CaseInsensitive(context)).forEach(printCommand);
+    reply += "```" + `\n`;
+  }
+
+  function CaseInsensitive(input){
+    return input ? input.toUpperCase() : undefined;
+  }
+
+  function printCommand(command){
+    reply += "Command :: " + command.keyword + `\n`;
+    reply += " - " +command.helptext + `\n`
+    if(command.args){
+      command.args.forEach(printArg);
+    }
+    reply += `\n`
+  }
+
+  function printArg(arg){
+    reply += `\t* ` + "arg: " + (arg.index ? arg.index : arg.keyword) + `\n`;
+    reply += `\t  ` + arg.helptext + `\n`
+  }
+
+  msg.channel.send(reply);
+}
+
+
+function attachToBot(bot) {
+  registerHelp();
+
+  // Command Processing.
+  bot.on('message', msg => {
+    if (!msg.guild) return;
+
+    commands.forEach(command => {
+      if(msg.content.startsWith(command.keyword)){
+        var args = extractArgs(command, msg.content);
+        command.action(msg, args);
+        if(!command.preserve) {
+          msg.delete();
```

Comment:
I'd also like to see this logic reversed. A lot of our commands @ users, so i'd like them to be able to quickly see why they are getting pinged. The default should preserve the message, unless someone wants to hid the command.

Reply:
Makes sense to me. Fixed in commit 8956031dab6a1252d5aa0241e88522f6c953fe77